src/test/java/com.gigssandbox/GigServiceTest.java

```diff
@@ -0,0 +1,114 @@
+package com.gigssandbox;
+
+import com.gigssandbox.entities.Gig;
+import com.gigssandbox.entities.User;
+import com.gigssandbox.exceptions.NoAppropriateGigsException;
+import com.gigssandbox.services.GigService;
+import java.util.Calendar;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.GregorianCalendar;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.internal.util.reflection.Whitebox;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+public class GigServiceTest {
+    private GigService gigService;
+
+    @Before
+    public void setUp() {
+        this.gigService = new GigService();
+    }
+
+    @Test
+    public void shouldAddUserToAttendeesWhenUserWantsToJoinGig() {
+        User user = User.builder().username("slaves").build();
+        String headliner = "Attila";
+        Calendar gigDate = new GregorianCalendar(2018, Calendar.OCTOBER, 14);
+
+        Gig.Credentials credentials = new Gig.Credentials(headliner, gigDate);
+        Gig gig = Gig.builder().credentials(credentials).attendees(new HashSet<>()).build();
+        Map<Gig.Credentials, Gig> gigs = new HashMap<>();
+        gigs.put(credentials, gig);
+        Whitebox.setInternalState(gigService, "gigs", gigs);
+
```

Comment:
1) when test become large then use double empty lines to split givenWhenThen parts
2) Avoid code duplication in tests. Extract duplicated parts into setUp method in order to see what logic exactly is changed test by test. It should be very easy to distinguish tests difference

Reply:
but I use map not in all tests in this test class